src/DbUp.Downgrade/Helpers/EmbeddedDowngradeScriptFinder.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using DbUp.Engine;
+
+namespace DbUp.Downgrade.Helpers
+{
+    /// <summary>
+    /// Comparing Scripts and DowngradeScripts embedded in assembly.
+    /// </summary>
+    public class EmbeddedDowngradeScriptFinder : IDowngradeScriptFinder
+    {
+        private readonly DowngradeScriptsSettings _downgradeScriptsSettings;
+
+        /// <summary>
+        /// Creates EmbeddedDowngradeScriptFinder with DowngradeScriptsSettings.
+        /// </summary>
+        /// <param name="downgradeScriptsSettings">Pre-builded DowngradeScriptsSettings.</param>
+        public EmbeddedDowngradeScriptFinder(DowngradeScriptsSettings downgradeScriptsSettings)
+        {
+            if(downgradeScriptsSettings == null)
+            {
+                throw new ArgumentNullException("downgradeScriptsSettings");
+            }
+
+            _downgradeScriptsSettings = downgradeScriptsSettings;
+        }
+
+        public SqlScript GetCorrespondingDowngradeScript(SqlScript scriptToExecute, List<SqlScript> downgradeScripts)
```

Comment:
I'm actually not using embedded scripts in favour of files stored in system. 
There - DefaultDowngradeScriptFinder seems to be implemented straight forward.
On the other hand here, because it's not easy to "debug" this method on my mind (at least for me), is there any way it could be unit tested?

Reply:
`DefaultDowngradeScriptFinder` is used to match StaticScriptProvider. Will rename it to be more clear.

Reply:
I have added Acceptance tests that validates insertion of downgrade scripts as well as successful downgrade. I have never been using `FileSystemScriptProvider` before, but all tests confirmed that `DefaultDowngradeScriptFinder` is working perfectly fine with is as long as Up and Down folders has the same structure. I didn't find any issue what so ever in matching by name, generally it works like the `StaticScriptProvider`. You may have another look again and execute the test ;)

Reply:
I think the name is fine. I was just simply misssing some tests in that area. But now seems like they are in.
That saying - whole pull request seems fine to me.

Reply:
I've run those unit tests locally and they are fine.
I've also tested my project with your source code attached and that also works as expected.
(my code is using files from file system).

Reply:
👍 glad to head that. Will merge it tomorrow and will publish a new version to nuget.